### PR TITLE
spec: allow to select rel and media types

### LINKED ISSUE #46: Support setting accepted media-type on Preload & Fields pushes
I choose to view Vulcain as a possible REST maturity models level 4. So I'm wondering if it should be possible to specify accepted media-type on pushes / preloads via Preload and Fields in order to fit well with already mature REST API's that uses media type for content negotiation.

In all transparency we use that in [eZ Platform](https://ezplatform.com/), and it's a rather powerful concept as it allows integrators to extend certain api endpoints. But unlike graphQL where caller can ask for whatever, in this case someone would need to code alternative representations.

While it would have to be in a format that can work both in the header and query format, something like the following might be a way to do it:
```
GET /books/ HTTP/2
Preload: /member:application%2Fvnd.acme.externalmember+json/*/author
Fields: /author/familyName
```

Open questions:
- Handling of `/`, here shown encoded using `%2F` as possible measure to avoid conflicts with path
- If type should have to be repeated on `Fields:` or not if already specified. Here suggested to be optional however no strong opinion on that.
- Separator to use to show start of media type, here shows as `:`, however that is legal in URI's so maybe it will have to be something else, did not find any [MIME's](https://www.iana.org/assignments/media-types/media-types.xhtml) with this tough
- Should it also support MIME type _parameter_? Like `;charset=UTF-8`

### PR DESCRIPTION
**Pull request by dunglas**
* Allow to select specific media types as requested in #46 but also specific relation types and languages
* Allow to select all links by passing an empty string (best used with the new `rel` parameter
* Switch to the format defined in the [structured headers I-D](https://httpwg.org/http-extensions/draft-ietf-httpbis-header-structure.html#string)
* Add an "Implementation Status" section ([RFC6982](https://tools.ietf.org/html/rfc6982)
* Add an "Acknowledgment" section

@andrerom could you review this change? It should address the eZ Publish's use case (closes #46).
@evert as discussed during API Days Interface, this change to the spec should allow to cover the use cases also covered by your [Prefer-Push I-D](https://tools.ietf.org/html/draft-pot-prefer-push-01). Would you mind to take a look?

Thanks!

### CODE REVIEW

**Comment by evert** [comment]
This looks great! Definitely meets my use-case and then some. I also like that structured headers is used for this!

File: spec/vulcain.md [hunk 1]
```diff
@@ -373,4 +468,111 @@ registry defined in [@!RFC3864].
 
 A selector registry could also be added.
 
+# Implementation Status
+
+[RFC Editor Note: Please remove this entire seciton prior to publication as an RFC.]
```
Comment (GregoireHebert) [inline spec/vulcain.md:473]:
```suggestion
[RFC Editor Note: Please remove this entire section prior to publication as an RFC.]
```

**Comment by dunglas** [comment]
@andrerom unfortunately I think that it's fundamentally incompatible with the other goad of this change which is to add selection capabilities similar to what the `Prefer-Push` header allows. Maybe could we add another attribute for this use case (`overloadtype`? `overloadaccept`?), but I wonder if it's not too specific. Maybe could we do as in the `Web Linking` RFC and allow custom attributes as an extension. The eZ would be able to pass its own attribute for this use case?

**Comment by dunglas** [comment]
In 3e548228e95ee9eaae68a6b6db857716a059cf25 I added a clarification regarding content negotiation: servers should try to guess the best possible value for the `Accept` and `Accept-Language` request headers.

**Pull request merged**